mypy/checkexpr.py

```diff
@@ -2697,6 +2720,8 @@ def overload_arg_similarity(actual: Type, formal: Type) -> int:
             # subtyping algorithm if type promotions are possible (e.g., int vs. float).
             if formal.type in actual.type.mro:
                 return 2
+            elif formal.type.is_protocol and is_subtype(actual, erasetype.erase_type(formal)):
```

Comment:
I am not sure, but in view of https://github.com/python/mypy/issues/3656 I am reluctant to increase similarity of anything to protocols.